packages/assets-controllers/src/NftController.ts

```diff
@@ -573,14 +572,11 @@ export class NftController extends BaseController<NftConfig, NftState> {
       Pick<ApiNftContract, 'address'> &
       Pick<ApiNftContract, 'collection'>
   > {
-    const name = await this.getERC721AssetName(
-      contractAddress,
-      networkClientId,
-    );
-    const symbol = await this.getERC721AssetSymbol(
-      contractAddress,
-      networkClientId,
-    );
+    const [name, symbol] = await Promise.all([
+      this.getERC721AssetName(contractAddress, networkClientId),
+      this.getERC721AssetSymbol(contractAddress, networkClientId),
+    ]);
```

Comment:
So I believe the reason we didn't do this originally is that `Promise.all` will reject all promises in the array if one of them rejects. We were waiting until we could use ES2020 syntax to use `Promise.AllSettled` which perhaps now we can?

Reply:
Yep I believe as of https://github.com/MetaMask/core/pull/1069 we can now use es2020 syntax so can we change this to `Promise.allSettled`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

Reply:
It's true that `Promise.allSettled` is available.  But don't we want this to throw?  The existing behavior is to throw if `name()` or `symbol()` fails, and this PR maintains that behavior.  `Promise.all` will reject with the first one that fails, which also behaves the same.

Reply:
Hmmm yeah I suppose that is the current behavior. Since ERC721s can have one, both or neither of these values however I would expect we would want predicate the fetching of one of them on the presence of both of them

Reply:
i do want to relax the current requirements that erc721 and erc20 define symbol and name, since they're optional in those specs.  Think that will be more involved though.